### ISSUE TITLE
Wifi dongle Vilros (Realtek) not connecting to WAP

### ISSUE THREAD

**Issue by wh1stler** [recalbox/recalbox-os#128]
Spent about 24 hours straight trying to get this specific dongle to connect - the Pi sees the device but the WAP never shows it as connected and it will not received DHCP - specifics below:
# lsusb

Bus 001 Device 006: ID 0bda:8176
Bus 001 Device 005: ID 050d:065a
Bus 001 Device 004: ID 046d:c52b
Bus 001 Device 003: ID 0424:ec00
Bus 001 Device 002: ID 0424:9514
Bus 001 Device 001: ID 1d6b:0002

Device is 0bda:8176 - Realtek Semiconductor Corp. RTL8188CUS 802.11n WLAN Adapter
# lsmod

Module                  Size  Used by    Not tainted
8192cu                509620  0
# ifconfig -a

eth0      Link encap:Ethernet  HWaddr B8:27:EB:B7:F5:27
          inet addr:##.##.#.#  Bcast:0.0.0.0  Mask:255.0.0.0
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:2511 errors:0 dropped:0 overruns:0 frame:0
          TX packets:207 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:180776 (176.5 KiB)  TX bytes:26496 (25.8 KiB)

lo        Link encap:Local Loopback
          inet addr:127.0.0.1  Mask:255.0.0.0
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:476 errors:0 dropped:0 overruns:0 frame:0
          TX packets:476 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:35300 (34.4 KiB)  TX bytes:35300 (34.4 KiB)

wlan0     Link encap:Ethernet  HWaddr 00:13:EF:90:35:4D
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:142 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

Interfaces
# Configure Loopback

auto lo
iface lo inet loopback

auto eth0
iface eth0 inet static
address ##.##.#.##
netmask 255.0.0.0
gateway ##.##.#.#

auto wlan0
# allow-hotplug wlan0

iface wlan0 inet dhcp
wpa-conf /etc/wpa_supplicant/wpa_supplicant.conf
# ifup wlan0

udhcpc (v1.23.1) started
Sending discover...
Sending discover...
Sending discover...
No lease, failing
# 

last lines of dmesg
[  832.477864] usb 1-1.5: USB disconnect, device number 6
[  832.478559] rtw_cmd_thread: DriverStopped(0) SurpriseRemoved(1) break at line 482
[  836.037320] usb 1-1.5: new high-speed USB device number 7 using dwc_otg
[  836.138758] usb 1-1.5: New USB device found, idVendor=0bda, idProduct=8176
[  836.138776] usb 1-1.5: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[  836.138787] usb 1-1.5: Product: 802.11n WLAN Adapter
[  836.138798] usb 1-1.5: Manufacturer: Realtek
[  836.138809] usb 1-1.5: SerialNumber: 00e04c000001

 /lib/udev/rules.d/75-persistent-net-generator.rules is there - 
70-persistence-net-rules is not in the rules.d dir

I am using WPA2 PSK with AES encryption on the WAP side.  I appreciate any help!  Thanks for this project - it is indeed epic.

**Comment by digitalLumberjack** [comment]
Hi ;)
`ps -e | grep wpa` shows wpa-supplicant ?

`ifdown wlan0 && killall wpa-supplicant && wpa_supplicant -iwlan0 -c/etc/wpa_supplicant/wpa_supplicant.conf &`

What brings `ifup wlan0` ?

**Comment by wh1stler** [comment]
Hi - thanks again - 
 ps -e | grep wpa
  312 root     wpa_supplicant -iwlan0 -c/etc/wpa_supplicant/wpa_supplicant.conf
  373 root     grep wpa

# ifdown: interface wlan0 not configured
killall: wpa-supplicant: no process killed
         
# ifup wlan0
udhcpc (v1.23.1) started
Sending discover...
Sending discover...
Sending discover...
No lease, failing
[2]+  Done(1)                    ifdown wlan0 && killall wpa-supplicant && wpa_supplicant -iwlan0 -c/etc/wpa_supplicant/wpa_supplicant.conf

thanks!

**Comment by wh1stler** [comment]
Any other info on this?  I ended up picking up a Edimax adapter to give it a try but the one I received is based off the Realtek chipset and is not connecting to wifi as well.  Any logs i can look at to see if it is even trying to connect?

**Comment by woolyninja** [comment]
Quick question for people with this issue - are you using a router with a hidden SSID?

This was my problem... my fix was to add the following under the ssid, etc in the wpa_supplicant.conf

scan_ssid=1

Once I added that wifi connected just fine.  Hope that helps anyone having this issue (and the devs).

**Comment by wh1stler** [comment]
I've added that config bit and had the same result as well as broadcasting the SSID rather than hiding it.  The log file shows authentication issues when connecting to the wifi using the realtek chipped dongle - I replaced it with a Edup that is on the compatibility list - didn't change any configuration and it worked perfectly.  Something with the driver on the realtek dongle's as far as I can tell.
